game/graphics/gfx.cpp

```diff
@@ -53,36 +52,162 @@ GfxGlobalSettings g_global_settings;
 GfxSettings g_settings;
 // const std::vector<const GfxRendererModule*> renderers = {&moduleOpenGL};
 
-// TODO serialize
-void LoadSettings() {
-  const auto filename = file_util::get_file_path({GAME_CONFIG_DIR_NAME, SETTINGS_GFX_FILE_NAME});
-  if (fs::exists(filename)) {
-    // this is just wrong LOL
-    FILE* fp = file_util::open_file(filename.c_str(), "rb");
-    lg::info("Found graphics configuration file. Checking version.");
-    u64 version;
-    fread(&version, sizeof(u64), 1, fp);
-    if (version == GfxSettings::CURRENT_VERSION) {
-      fseek(fp, 0, SEEK_SET);
-      fread(&g_settings, sizeof(GfxSettings), 1, fp);
-      lg::info("Loaded graphics configuration file.");
-    } else {
-      // TODO upgrade func
-      lg::info("Detected graphics configuration file from old version. Ignoring.");
+// Not crazy about this declaration
+const std::pair<std::string, Pad::Button> gamepad_map[] = {{"Select", Pad::Button::Select},
+                                                           {"L3", Pad::Button::L3},
+                                                           {"R3", Pad::Button::R3},
+                                                           {"Start", Pad::Button::Start},
+                                                           {"Up", Pad::Button::Up},
+                                                           {"Right", Pad::Button::Right},
+                                                           {"Down", Pad::Button::Down},
+                                                           {"Left", Pad::Button::Left},
+                                                           {"L1", Pad::Button::L1},
+                                                           {"R1", Pad::Button::R1},
+                                                           {"Triangle", Pad::Button::Triangle},
+                                                           {"Circle", Pad::Button::Circle},
+                                                           {"X", Pad::Button::X},
+                                                           {"Square", Pad::Button::Square}};
+
+const std::pair<std::string, Pad::Analog> analog_map[] = {
+    {"Left X Axis", Pad::Analog::Left_X},
+    {"Left Y Axis", Pad::Analog::Left_Y},
+    {"Right X Axis", Pad::Analog::Right_X},
+    {"Right Y Axis", Pad::Analog::Right_Y},
+};
+
+void DumpToJson(const std::string& filename) {
```

Comment:
we've been trying to use `filesystem::path` when possible for filenames. I think you can just switch this argument to use `filesystem::path`

Reply:
Ok, I'll update this when I get a chance